src/hgvs/utils/altseq_to_hgvsp.py

```diff
@@ -190,10 +191,17 @@ def _convert_to_sequence_variants(self, variant, acc):
             self._is_ambiguous = True  # side-effect
 
         if insertion and insertion.find("*") == 0:  # stop codon at variant position
-            aa_start = aa_end = AAPosition(base=start, aa=deletion[0])
-            ref = ""
-            alt = "*"
-            is_sub = True
+            if start == len(self._alt_seq) and not deletion:
+                # insertion of stop codon at end of sequence, a special case of "stop_retained"
+                aa_start = aa_end = AAPosition(base=start, aa="*")
+                ref = "*"
+                alt = "*"
+                is_sub = True
+            else:
+                aa_start = aa_end = AAPosition(base=start, aa=deletion[0])
+                ref = ""
+                alt = "*"
+                is_sub = True
```

Comment:
Since `is_sub` and `alt` are the same regardless of the condition, we can move them outside of the if statement
```suggestion
            is_sub = True
            alt = "*"
            if start == len(self._alt_seq) and not deletion:
                # insertion of stop codon at end of sequence, a special case of "stop_retained"
                aa_start = aa_end = AAPosition(base=start, aa="*")
                ref = "*"
            else:
                aa_start = aa_end = AAPosition(base=start, aa=deletion[0])
                ref = ""
```